extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class ContributionAppreciationController extends WikiaController {
+
+	public function appreciate() {
+		global $wgUser;
+
+		if ( $this->request->wasPosted() && $wgUser->matchEditToken( $this->getVal( 'token' ) ) ) {
+			//TODO: do something with apprecation
+			$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+			$this->setVal( 'user', $wgUser->getName() );
+			$this->setVal( 'for', Revision::newFromId( $this->request->getInt( 'revision' ) )->getUserText() );
```

Comment:
can we make this val more descriptive than `for` ? `for` can be both receiver of the appreciation and the thing he made so it can be difficult to figure out outside of this context here